### PR TITLE
Creation of tables for Acc, Cust and Transaction

### PR DESCRIPTION
**Pull request by RoyTeoh**

### CODE REVIEW

File: capstone_db.session.sql [hunk 1]
```diff
@@ -6,4 +6,8 @@
 --         '$2a$10$qLawnDP..8UX.KvfKZ9jBe1MZt1KR6bob/rs.111v8FJcPGYQ/qE.',
 --         TRUE
 --     );
--- INSERT INTO users_roles (role_id, user_id) VALUES (1, 1)
\ No newline at end of file
+-- INSERT INTO users_roles (role_id, user_id) VALUES (1, 1)
+
+ALTER TABLE accounts
+ADD CONSTRAINT check_status
+CHECK (status IN ('active', 'frozen', 'deleted'));
```
Comment (majulahsingapuri) [inline capstone_db.session.sql]: Just comment these out

File: src/main/java/com/example/capstone3/Account.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        this.customer = customer;
+    }
+
+    public String getStatus() {
+        return this.status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }}
```
Comment (majulahsingapuri) [inline src/main/java/com/example/capstone3/Account.java]: enforce status checking here as well. There is an [article](https://www.baeldung.com/jpa-persisting-enums-in-jpa) about persisting enums. you should take a look at that and implement accordingly

File: src/main/java/com/example/capstone3/Customer.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package com.example.capstone3;
+
+import java.util.ArrayList;
+import java.util.List;
```
Comment (majulahsingapuri) [inline src/main/java/com/example/capstone3/Customer.java]: Remove unused imports

File: src/main/java/com/example/capstone3/Transaction.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    private LocalDate datecreated;
+    private String type;
+    private double amount;
+
+
+    public Transaction() {
+        super();
+    }
+
+    public Transaction(Long id, String username, String password, Account account) {
```
Comment (majulahsingapuri) [inline src/main/java/com/example/capstone3/Transaction.java]: Incorrect constructor definition

**Review by majulahsingapuri — DISMISSED** [review]
Some glaring errors need to be fixed

File: src/main/java/com/example/capstone3/Account.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+public class Account {
+
+    @Id
+    @Column(name = "account_id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long accountId;
+    private String name;
+    private String accountType;
+
+       public Account() {
```
Comment (helmyrsln) [inline src/main/java/com/example/capstone3/Account.java:27]: add super()

File: src/main/java/com/example/capstone3/Customer.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    @Id
+    @Column(name = "customer_id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long customerId;
+
+    private String firstname;
+    private String lastname;
+
+
+    public Customer() {
```
Comment (helmyrsln) [inline src/main/java/com/example/capstone3/Customer.java:16]: add super()

**Review by helmyrsln — DISMISSED** [review]

**Review by Jus323 — APPROVED** [review]

**Review by majulahsingapuri — APPROVED** [review]

**Review by helmyrsln — APPROVED** [review]

**Pull request merged**